common/utils/dates.ts

```diff
@@ -36,6 +36,16 @@ export function isSameDay(date1: Date, date2: Date): boolean {
   return isSameMonth(date1, date2) && date1.getUTCDate() === date2.getUTCDate();
 }
 
+// Note: the order of arguments to this function is designed so you can
+// concatenate them and get sensible-looking results.
+//
+//      isSameDayOrBefore(A, B) && isSameDayOrBefore(B, C)
+//        => isSameDayOrBefore(A, C)
+//
+export function isSameDayOrBefore(date1: Date, date2: Date): boolean {
+  return isSameDay(date1, date2) || date1 <= date2;
+}
```

Comment:
These are two helpers that will be useful for rewriting `@weco/catalogue/utils/dates` in terms of `Date` instead of `Moment`. They aren't used yet, but this gets them in place so we can just use them in future patches.